Lib/gettext.py

```diff
@@ -491,11 +492,16 @@ def find(domain, localedir=None, languages=None, all=False):
         localedir = _default_localedir
     if languages is None:
         languages = []
-        for envar in ('LANGUAGE', 'LC_ALL', 'LC_MESSAGES', 'LANG'):
-            val = os.environ.get(envar)
-            if val:
-                languages = val.split(':')
-                break
+        if os.environ.get('LANGUAGE'):
+            languages = os.environ.get('LANGUAGE').split(',')
+        elif locale.getlocale() != (None, None):
+            languages.append(".".join(filter(None, locale.getlocale())))
```

Comment:
I'd avoid calling `getlocale()` twice here. Maybe you could do something like this?
```python
elif (loc := locale.getlocale()) != (None, None):
    ...
```